joblib/test/conftest.py

```diff
@@ -0,0 +1,6 @@
+from joblib.testing import fixture
+
+
+@fixture(scope='function')
```

Comment:
Just checking we don't actually need to write a fixture, provided we use `tmpdir` as the test function arguments and `tmpdir.strpath` in the test function body right?

Maybe this solution would have my preference.